### PR TITLE
automatically setup etcd when running {verify,update}-openapi-spec.sh

### LINKED ISSUE #121492: dockerized verify-openapi-spec.sh doesn't work due to etcd install
we just log that the user should install etcd, which is impractical in an ephemeral container

we actually already call installing etcd in the verify script, but we don't export it into the current PATH for the process so it doesn't work ...

ref: https://github.com/kubernetes/kubernetes/issues/117821#issuecomment-1777862886

/sig api-machinery
/assign
/kind bug

### PR DESCRIPTION
**Pull request by BenTheElder**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

1. Automatically add etcd to current PATH when calling kube::etcd::install

2. Call kube::etcd::install from update-openapi-spec

Ensures that e.g. `build/run.sh make verify WHAT=openapi-spec` (or just `make verify`) actually works without the user having to manually run the "install etcd into third_party" steps.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/121492
/sig api-machinery
/hold

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/lib/etcd.sh [hunk 1]
```diff
@@ -144,38 +144,40 @@ kube::etcd::cleanup() {
 }
 
 kube::etcd::install() {
-  (
```
Comment (BenTheElder) [inline hack/lib/etcd.sh:147]: this is de-indented and dropping the `()` subshell, then adding the two `export ...` lines

Comment (thockin) [reply hack/lib/etcd.sh:147]: we call `cd` later - that changes the working dir for the caller.  This seems broken

Comment (thockin) [reply hack/lib/etcd.sh:147]: yeah, I checked out a brand new tree at HEAD and I hit the path which returns from this function with `pwd` in the third_party directory

Comment (thockin) [reply hack/lib/etcd.sh:147]: bash is super wonky about exit codes and errexit and conditionals, but I am cooking something up

Comment (thockin) [reply hack/lib/etcd.sh:147]: https://github.com/kubernetes/kubernetes/pull/122470

Comment (BenTheElder) [reply hack/lib/etcd.sh:147]: > we call cd later - that changes the working dir for the caller. This seems broken

agree. thanks for fixing.

**Pull request merged**